src/SFML/Window/DRM/VideoModeImpl.cpp

```diff
@@ -53,7 +53,12 @@ std::vector<VideoMode> VideoModeImpl::getFullscreenModes()
 ////////////////////////////////////////////////////////////
 VideoMode VideoModeImpl::getDesktopMode()
 {
-    if (const auto* ptr = DRMContext::getDRM().mode)
+    const Drm&         drm = DRMContext::getDRM();
+    drmModeModeInfoPtr ptr = drm.mode;
+    if (!ptr) // if no mode has been set, return the original mode we started with
+        ptr = &(drm.originalCrtc->mode);
+
+    if (ptr)
```

Comment:
At this point it's not possible for `ptr` to be null. `&(drm.originalCrtc->mode)` has to return a non-null address. You can remove this conditional.